### PR TITLE
Jmoser edit pins

### LINKED ISSUE #83: Edit Pin component
We need a way to edit Pins after they are added.  Probably an extension of the PinPage component that is blocked by role.

### PR DESCRIPTION
**Pull request by stretch1414**
Fixes #83

### CODE REVIEW

File: src/components/CollectionSelectList.js [hunk 1]
```diff
@@ -172,10 +172,10 @@ class CollectionSelectList extends Component {
             });
         });
         if (collectionToAdd.length > 0) {
-            this.addToCollection(collectionToAdd, collections);
+            this.addToCollection(collectionToAdd[0], collections);
         }
         if (collectionToDeleteFrom.length > 0) {
-            this.deleteFromCollection(collectionToDeleteFrom, collections);
+            this.deleteFromCollection(collectionToDeleteFrom[0], collections);
         }
```
Comment (markmoser) [inline src/components/CollectionSelectList.js]: What happens if the `collectionToDeleteFrom` array has > 1 entry? Should this iterate and remove all? I'm probably missing some context here.

Comment (stretch1414) [reply src/components/CollectionSelectList.js]: So it is not exactly possible with the way that the select list box works currently.  Though I'm thinking about refactoring that a bit for a different reason (that escapes me now).

Comment (markmoser) [reply src/components/CollectionSelectList.js]: I experienced the UI today and think I understand the intention better. What happens if someone has a pin in multiple categories, doe sit always remove from the first collection in the list?

Testing this I added two collections, added a pin to both, and the pin is not removed from either collection on refresh.

Comment (stretch1414) [reply src/components/CollectionSelectList.js]: It should be treating any addition or removal as an independent action. But we may have a race condition problem. May need a write lock on that. Unless I'm misunderstanding the issue you ran into?

**Comment by scytherswings** [comment]
So long story short, I was going to get Joel's edit page in, but then I found out that @markmoser 's changes to the way we request images caused some newer versions of chrome to block mixed http/https, which made me go down the rabbithole of https://github.com/PinsterTeam/ImageService/pull/52 So once that merges we will be safe to push the UI to prod, I think. Though that's going to be a migration with some downtime for sure.

**Pull request merged**